benchmarking/leaf_node.py

```diff
@@ -0,0 +1,202 @@
+from parameter import Parameter
+import os
+import csv
+import random
+import string
+import numpy as np
+import uuid
+from concurrent.futures import ThreadPoolExecutor
+from concurrent.futures import as_completed
+import time
+import threading
+
+class LeafNode:
+    """Represents the leaf nodes of the threaded tree.
+    
+    Responsible for generating the data set of message content which conforms 
+    to the parameter values. 
+    Responsible for sending requests to the createMessage API.
+
+    Attributes:
+        parameter_to_value: A dictionary mapping parameter names to values for all 
+                            parameters.
+    """
+   
+    def __init__(self, parameter_to_value):
+        """Initializes Leaf Node with parameter_to_value."""
+
+        self.parameter_to_value = parameter_to_value.copy() 
+        
+    def __get_qps(self):
+        return self.parameter_to_value["qps"]
+
+    def __get_prob_of_file(self):
+        return self.parameter_to_value["prob_of_file"]
+
+    def __get_mean_length_of_text(self):
+        return self.parameter_to_value["mean_length_of_text"]
+
+    def __get_std_dev_length_of_text(self):
+        return self.parameter_to_value["std_dev_length_of_text"]
+
+    def __get_mean_length_of_text_with_file(self):
+        return self.parameter_to_value["mean_length_of_text_with_file"]
+
+    def __get_std_dev_length_of_text_with_file(self):
+        return self.parameter_to_value["std_dev_length_of_text_with_file"]
+
+    def __get_mean_file_size(self):
+        return self.parameter_to_value["mean_file_size"]
+
+    def __get_std_dev_file_size(self):
+        return self.parameter_to_value["std_dev_file_size"]
+
+    def __sample_normal_file_size_generate_random_binary_file(self, mean_file_size, std_dev_file_size):
+        file_size = min(25, 
+                        round(np.random.normal(
+                                loc = mean_file_size, 
+                                scale = std_dev_file_size
+                                )
+                            )
+                        )
+        file_name = os.path.join(
+                        Parameter.DIRECTORY, 
+                        "{}_{}_{}_".format(
+                                    file_size, 
+                                    mean_file_size, 
+                                    std_dev_file_size
+                                    )
+                            + str(uuid.uuid4())
+                        )
+        with open('%s'%file_name, 'wb') as fout:
```

Comment:
So you are creating a file, writing random data to it. Then you are reading from it and sending the request.

1. You are not actually deleting the file. This will end up creating numerous files on your device.
2. You don't actually need the file. All you need is the bytes contained in it. So you can just have this function return the bytes instead of actually creating a file and reading from it.

Reply:
Done